### DISCUSSION TITLE
Syncing PaperMerge with other cloud systems

### DISCUSSION THREAD

**Discussion by vbaError** [papermerge/papermerge-core#19] [Q&A]
**Is there a way to sync PaperMerge with other cloud systems like Synology Drive or Google Drive** to leverage the strength of each when storing, organizing, and accessing the documents?

I was happy to find Papermerge because it's so easy to organize my documents with metadata. Papermerge impressed me because it does a lot with no setup, unlike other software I have been using.

I currently use a combination of VBA (FileSystemObject), ImageMagick, AutoHotkey, Ms. Access, and Synology Drive to store, organize and retrieve documents quickly, and I would like to use PaperMerge in my workflow.

**Comment by ciur** [comment]
Syncronization (whatever is meant by that) is not provided by core features.
However, you can leverage [rich REST API](https://docs.papermerge.io/redoc/) to copy/sync documents (their versions, tags, folder structure etc) with whatever 3rd party system you may think of.
This basically means that there is no button, menu or some script to sync with say Google Drive. But, if you know how to use [curl](https://curl.se/), python, ruby, bash or whatever library/language which provides you an http client - you can do it via REST API.

Please mind that both development and REST API documentation is still work in progress :)

Thank you for opening discussion on this topic. I will keep this discussion open to remind myself to write documentation/tutorials on how to create a simple client to sync with say google drive/dropbox.

**Reply by vbaError** [reply]
Thank you for your answer.  I will explore the Rest Api through msxml2.xmlhttp, the corresponding VBA library. 

When I mentioned Synchronization, I mostly thought about two-way synchronization to backup or sync my data between cloud services.  As a Synology user, I was thinking of something similar to Synology's cloud sync as described here https://www.synology.com/en-us/dsm/feature/cloud_sync

The idea is to be able to use the best of PaperMerge without giving up some of the robust features provided by Synology's Cloud Services that I already use and depend on. In my case Synology Drive: https://www.synology.com/en-us/dsm/feature/drive

![image](https://user-images.githubusercontent.com/104432665/165854217-37b46728-e7e8-4146-951d-26d7d729c860.png)
